infra/Pos/Network/Types.hs

```diff
@@ -142,8 +142,8 @@ topologySubscriptionWorker = go
 topologyRunKademlia :: Topology kademlia -> Maybe kademlia
 topologyRunKademlia = go
   where
-    go (TopologyRelay _ kademlia)         = Just kademlia
-    go (TopologyP2P _ _ kademlia)         = Just kademlia
+    go (TopologyRelay _ mKademlia)        = mKademlia
```

Comment:
The docs say it defaults to False for core, but that it can be turned on. To make that true we'd need a case here for `TopologyCore`.

BTW, can we get rid of `go` here and promote the pattern matching to the top?